### PR TITLE
Fixes #18300: upgrade testing loop to match new actions and dsc integration

### PR DESCRIPTION
**Pull request by gpoblon**
https://issues.rudder.io/issues/18300

### CODE REVIEW

File: rudder-lang/src/bin/rudderc.rs [hunk 1]
```diff
@@ -88,7 +88,7 @@ fn main() {
     };
     let command_status = command_result.is_ok();
```
Comment (amousset) [inline rudder-lang/src/bin/rudderc.rs:89]: naming it `command_success` or something like that would have prevented the mistake.

Comment (gpoblon) [reply rudder-lang/src/bin/rudderc.rs]: True, fixuped. And added an explicit error for cli errors

**Comment by amousset** [comment]
OK for me, I'll let @peckpeck have a look

**Review by peckpeck — APPROVED** [review]

**Comment by gpoblon** [comment]
OK, squash merging this PR

**Comment by gpoblon** [comment]
OK, merging this PR

**Pull request merged**